runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8004,35 +8007,87 @@ class INTERPRETER_CLASS
 				goto retry;
 			}
 		}
+
 		return rc;
-#else
-	Assert_VM_unreachable();
-	return EXECUTE_BYTECODE;
-#endif /* J9VM_OPT_METHOD_HANDLE */
 	}
+#elif defined(J9VM_OPT_OPENJDK_METHODHANDLE) /* defined(J9VM_OPT_METHOD_HANDLE) */
+	VMINLINE VM_BytecodeAction
+	invokedynamic(REGISTER_ARGS_LIST)
+	{
+retry:
+		VM_BytecodeAction rc = GOTO_RUN_METHOD;
+		U_16 index = *(U_16 *)(_pc + 1);
+
+		J9ConstantPool *ramConstantPool = J9_CP_FROM_METHOD(_literals);
+		J9InvokeCacheEntry *invokeCache = (J9InvokeCacheEntry *)ramConstantPool->ramClass->callSites + index;
```

Comment:
Maybe put the casting in brackets? This help to make the pointer addition target more clear
`((J9InvokeCacheEntry *)ramConstantPool->ramClass->callSites) + index;`

Reply:
Updated.